src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -284,91 +275,149 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
+        break;
+      case test_mode::support:
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Checking Test Support", MAX_TEST_DURATION, os_stdout, os_stderr);
+        break;
+      case test_mode::no_json:
+        if (_ptTest.get<std::string>("file").find(".exe") != std::string::npos)
+          exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
+        else
+          exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
+        break;
+      default:
+        throw std::runtime_error("Invalid test mode specified"); //Mode was not specified correctly.
+    }
+
+    if (exit_code == EOPNOTSUPP) {
+      logger(_ptTest, "Details", "Test is not supported on current device.");
+      _ptTest.put("status", test_token_skipped);
+      return false;
+    } else if (exit_code == EXIT_SUCCESS) {
+      if (mode != test_mode::support)
         _ptTest.put("status", test_token_passed);
-      }
-      else {
-        logger(_ptTest, "Error", os_stdout.str());
-        logger(_ptTest, "Error", os_stderr.str());
-        _ptTest.put("status", test_token_failed);
-      }
-    } catch (const std::exception& e) {
-      logger(_ptTest, "Error", e.what());
+      return true;
+    } else {
+      logger(_ptTest, "Error", os_stdout.str());
+      logger(_ptTest, "Error", os_stderr.str());
       _ptTest.put("status", test_token_failed);
+      return false;
     }
+  } catch (const std::exception& e) {
+    logger(_ptTest, "Error", e.what());
+    _ptTest.put("status", test_token_failed);
+    return false;
   }
-  else {
-    //check if testcase is present
-    std::string xrtTestCasePath = "/opt/xilinx/xrt/test/" + py;
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+  return false;
+}
 
-    std::vector<std::string> args = { "-k", xclbinPath,
-                                      "-d", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(_dev)) };
-    int exit_code;
-    try {
-      if (py.find(".exe") != std::string::npos)
-        exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
-      else
-        exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
+/*
+ * Helper function to query support for black-box tests or run them.
+ */
+bool
```

Comment:
Why does this method need to return its functional status.  This is C++ not C code.